### DISCUSSION TITLE
Difference between MagickGetImageInterlaceScheme and MagickGetInterlaceScheme

### DISCUSSION THREAD

**Discussion by Danack** [ImageMagick/ImageMagick#4171] [Development] [answered]
Hi, 

Please could you clarify the difference between [MagickGetImageInterlaceScheme](https://github.com/ImageMagick/ImageMagick/blob/0b295fbcf5d69a4388f67df55a7dbd468f5786e8/MagickWand/magick-image.c#L5381-L5394) and [MagickGetInterlaceScheme](https://github.com/ImageMagick/ImageMagick/blob/0b295fbcf5d69a4388f67df55a7dbd468f5786e8/MagickWand/magick-property.c#L965-L972)

From reading the descriptions of these two functions it sounds like they should do the same thing. However it seems for an interlaced PNG image the values are:

```
getImageInterlaceScheme: 7
getInterlaceScheme: 1
```

I can see that internally the code is looking at a different data structure, but I don't know the difference in meaning.
```
MagickGetImageInterlaceScheme - return(wand->images->interlace);

MagickGetInterlaceScheme(MagickWand *wand) - return(wand->image_info->interlace);
```

**Comment by dlemstra** [comment, marked as answer]
The `MagickGetInterlaceScheme` returns the interlace setting from the "options". In other words the `-interlace` option on the command line. And `MagickGetImageInterlaceScheme` returns the interlace that was read from the image information. I hope that answers your question?

**Reply by Danack** [reply]
Thanks. I think it does.
